src/science/canopy/cbl_latent_heat.F90

```diff
@@ -82,34 +192,50 @@ SUBROUTINE Latent_heat_flux( mp, CTFRZ, dels, soil_zse, soil_swilt,           &
 
      canopy_fess(j) = MIN(canopy_fess(j), REAL(fupper_limit(j),r_2))
 
-     !Ticket 137 - case iii)
-     !evaporation from frozen soils needs to respect the assumption that
-     !ice fraction of soil moisture cannot exceed frozen_limit=0.85
-     !see soilsnow: if frozen_limit changes need to be consistent
+!|     - Limit 2: Additionally for case 3, the evaporation of liquid water
+!        from within the frozen soil column must not reduce the liquid water fraction
+!        to the point that that ice fraction of soil moisture exceeds an upper limit
+!        frozen_limit=0.85.  This provides a second upper limit on the evaporation and
+!        soil latent flux. **WARNING** frozen_limit=0.85 is hard coded - if it is changed
+!        then the corresponding limit in [[cbl_soilsnow]] must also be changed.
+!
      fupper_limit(j) = REAL(ssnow_wb(j)-ssnow_wbice(j)/0.85)*frescale(j)
      fupper_limit(j) = MAX(REAL(fupper_limit(j),r_2),0.)
 
      canopy_fess(j) = MIN(canopy_fess(j), REAL(fupper_limit(j),r_2))
 
-  END IF
-
+  END IF   
+!*     - The case of dew fall onto a surface with little/no snow while
+!        the soil surface temperature is above freezing is unrestricted and
+!        the soil latent heat flux defaults to the first estimate.
+  
   ssnow_cls(j)=1.
 
-  !Ticket 137 - case ii) deposition of frost onto snow
-  ! case of sublimation of snow overwrites later
+!|     - If the surface has snow cover, or `ssnow_potev`<0 **and** the soil
+!        temperature is below freezing (frost), then the latent heat flux represents a
+!        conversion between solid and vapour phases of water, \(c_{ls}\)=1.1335.
+!        The first estimate for the soil latent heat flux is updated by the change
+!        in value of \(c_{ls}\).
+!
   IF (ssnow_snowd(j) >=0.1 ) THEN
      ssnow_cls(j) = 1.1335
      canopy_fess(j) = ssnow_cls(j)*ssnow_potev(j)
```

Comment:
This has nothing to do with the documentation. Why isn't there "ssnow_wetfac" here to calculate canopy_fess?

Reply:
ssnow_wetfac quantifies the role that the moisture in the surface layer of the soil has in setting evaporation.  It lies between 0 and 1 (i.e. 0 if there's insufficient water to permit any evaporation, to 1 if there's sufficient water in the soil that it behaves like a lake).  With a snow layer, the sublimation flux comes from the snow pack - and this is sully wet (i.e. there's is no limitation).  Given that ssnow_wetfac may not be evaluated properly when snow is present the code just enforces ssnow_wetfac = 1 when calculating canopy_fess.

Importantly canopy_fess is what controls the change in water from the soil/snow pack over the time step - not ssnow_wetfac.

Reply:
OK. I see. And I see how to make this clearer. To me, it would be clearer if ssnow_wetfac and ssnow_cls were set correctly for all points first. Then have: `canopy_fess = ssnow_wetfac*ssnow_cls*ssnow_potev`. Then have the loop to apply the various limits. So this case and the next one wouldn't be "special" cases but simply the normal equation. It's possible to avoid resetting these values by setting the special cases correctly in the loop.

Or is it a problem to reset ssnow_wetfac to 1 for these points? Or is it a problem not to reset it? Because ssnow_wetfac is used afterwards in the code and without the `if` conditions. So for some points with snow, we enforce ssnow_wetfac=1 here but use a different value later in the code (the one that was maybe wrongly evaluated by CABLE). For example, line 808 in cable_canopy.F90 and after, there is no snow case there.

Reply:
This section of code has evolved over a period of time with individuals making incremental changes - it's likely that a rewrite/restructure is worth it.  In this case what you suggest appears sensible - however we'd have to check against the various uses/setting of ssnow%wetfac - in define_canopy and soil_snow.  

The place you've identified (qscreen) isn't too much of an issue (a diagnostic that very few people look at).   

The bigger issue is likely the use of wetfac when setting the sensitivity terms (dfe_ddq) and what happens in the case of a change of sign of fes during the time step.  Again, unlikely to be much of an issue for most researchers - but is of relevance to ACCESS.  At the moment I think the code is buggy but internally consistent - changing things to fix the bug(s) could make consistency hard to achieve and might cause problems (or may not, it could solve the remaining mass conservation quirks).

Possibly worth a comment in the docs - that code will have to be looked at for JAC anyway.,

Reply:
OK. Got it.